### PR TITLE
Enable salesforce listener tests

### PR DESCRIPTION
**Pull request by aashikam**
# Description
fixes: https://github.com/ballerina-platform/ballerina-library/issues/6909
$title

### CODE REVIEW

File: ballerina/tests/listener_test.bal [hunk 1]
```diff
@@ -17,79 +17,81 @@
...
-//     auth: {
-//         username: username,
-//         password: password
-//     }
-// };
-// listener Listener eventListener = new (listenerConfig);
+ListenerConfig listenerConfig = {
+    auth: {
+        username: username,
+        password: password
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:29]: Can't we do this ?
```suggestion
        username,
        password
```

File: ballerina/tests/listener_test.bal [hunk 2]
```diff
@@ -17,79 +17,81 @@
...
-//             io:println("Restored " + payload.toString());
-//         } else {
-//             io:println(payload.toString());
-//         }
-//     }
-// }
+service "/data/ChangeEvents" on eventListener {
+    remote function onCreate(EventData payload) {
+        string? eventType = payload.metadata?.changeType;
+        if (eventType is string && eventType == "CREATE") {
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:42]: I think we can do this.
```suggestion
        if eventType is  "CREATE" {
```

File: ballerina/tests/listener_test.bal [hunk 3]
```diff
@@ -17,79 +17,81 @@
...
+            }
+            io:println("Created " + payload.toString());
+        } else {
+            io:println(payload.toString());
+        }
+    }
+
+    remote isolated function onUpdate(EventData payload) {
+        json accountName = payload.changedData.get("Name");
+        if (accountName.toString() == "WSO2 Inc") {
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:54]:
```suggestion
        if accountName.toString() == "WSO2 Inc" {
```

File: ballerina/tests/listener_test.bal [hunk 4]
```diff
@@ -17,79 +17,81 @@
...
+                isCreated = true;
+            }
+            io:println("Created " + payload.toString());
+        } else {
+            io:println(payload.toString());
+        }
+    }
+
+    remote isolated function onUpdate(EventData payload) {
+        json accountName = payload.changedData.get("Name");
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:53]: Can't we do this ?
```suggestion
        string accountName = check payload.changedData.get("Name").ensureType();
```

File: ballerina/tests/listener_test.bal [hunk 5]
```diff
@@ -17,79 +17,81 @@
...
+            }
+            io:println("Updated " + payload.toString());
+        } else {
+            io:println(payload.toString());
+        }
+    }
+
+    remote function onDelete(EventData payload) {
+        string? eventType = payload.metadata?.changeType;
+        if (eventType is string && eventType == "DELETE") {
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:66]:
```suggestion
        if eventType is "DELETE" {
```

File: ballerina/tests/listener_test.bal [hunk 6]
```diff
@@ -17,79 +17,81 @@
...
+            }
+            io:println("Deleted " + payload.toString());
+        } else {
+            io:println(payload.toString());
+        }
+    }
+
+    remote function onRestore(EventData payload) {
+        string? eventType = payload.metadata?.changeType;
+        if (eventType is string && eventType == "UNDELETE") {
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:78]:
```suggestion
        if eventType is "UNDELETE" {
```

File: ballerina/tests/listener_test.bal [hunk 7]
```diff
@@ -17,79 +17,81 @@
...
+    }
+}
 
 // Using direct-token config for client configuration
 Client lisbaseClient = check new (sfConfigRefreshCodeFlow);
 string testRecordId = "";
 
 @test:Config {
-    enable: false
+    enable: true
```
Comment (ayeshLK) [inline ballerina/tests/listener_test.bal:94]: Do we need to manually do this ? I think this is enabled by default :thinking:

File: ballerina/tests/listener_test.bal [hunk 8]
```diff
@@ -17,79 +17,81 @@
 import ballerina/lang.runtime;
 import ballerina/log;
 import ballerina/test;
+import ballerina/os;
+import ballerina/io;
```
Comment (NipunaRanasinghe) [inline ballerina/tests/listener_test.bal:21]: Imports are not properly ordered. Use ballerina formatter

File: ballerina/tests/listener_test.bal [hunk 9]
```diff
@@ -17,79 +17,81 @@
...
-//                 isRestored = true;
-//             }
-//             io:println("Restored " + payload.toString());
-//         } else {
-//             io:println(payload.toString());
-//         }
-//     }
-// }
+service "/data/ChangeEvents" on eventListener {
+    remote function onCreate(EventData payload) {
```
Comment (NipunaRanasinghe) [inline ballerina/tests/listener_test.bal:40]:
```suggestion
service "/data/ChangeEvents" on eventListener {

    remote function onCreate(EventData payload) {
```

File: ballerina/tests/listener_test.bal (hunk 7 above)
Comment (aashikam) [reply ballerina/tests/listener_test.bal:94]: no these are currently not running due to an auth error. 
Secret mixup which I sorted out with Maheshika. email " Adding GitHub Secrets to the Ballerina Salesforce Repository"

File: ballerina/tests/listener_test.bal [hunk 10]
```diff
@@ -108,7 +110,7 @@ function testCreateRecord() {
 }
 
 @test:Config {
-    enable: false,
+    enable: true,
```
Comment (NipunaRanasinghe) [inline ballerina/tests/listener_test.bal:113]: No need to keep the `enable` property if the test is enabled. Apply where necessary
```suggestion
```

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**